src/server/system_services/account_server.js

```diff
@@ -1085,8 +1086,42 @@ async function check_google_connection(params) {
     }
 }
 
+async function check_aws_sts_connection(params) {
```

Comment:
again maybe instead of this code duplication -  just create the needed s3_params in both cases of aws and then send to the original check_aws_connection

Reply:
Done.